x-pack/plugins/security_solution/public/flyout/document_details/right/components/graph_preview_container.tsx

```diff
@@ -28,15 +27,20 @@ export const GraphPreviewContainer: React.FC = () => {
     ecsData: dataAsNestedObject,
   });
 
+  const timestamp = getField(getFieldsData('@timestamp'));
+
   // TODO: default start and end might not capture the original event
-  const graphFetchQuery = useFetchGraphData({
+  const { isLoading, isError, data } = useFetchGraphData({
     req: {
       query: {
         eventIds,
-        start: DEFAULT_FROM,
-        end: DEFAULT_TO,
+        start: moment(timestamp).subtract(30, 'minutes').toISOString(),
+        end: moment(timestamp).add(30, 'minutes').toISOString(),
```

Comment:
What happens if `timestamp` is `null`? In lines 37-38, `start` and `end` will be set to `null`. 

Should we default to something like i.e. `'now-30m/m'` and `'now+30m/m'` or is it safe to leave them as `null`?

Reply:
Great question. This component is not rendered if the timestamp is missing. The same is true when eventIds is undefined.

These checks happen on the parent component. This design is not ideal. I hope to find a better flow and refactor it. I'm also open to ideas.

and I'll run a jest UT to check what happens when timestamp is null; i'm curious myself

Reply:
should a new unit test be added to check that?

Reply:
Having to think about the side effect of `moment(undefined)` leading to the current Timestamp, seems to add to me some cognitive load that I would like to avoid it possible, i.e instead of falling back to undefined we can explicitly fallback to the current date object.

Also, I liked @albertoblaz suggestion of using Elasticsearch relatives start/end, so I guess we could combine both solutions in this way for ex:

```js
  const timestamp = getField(getFieldsData('@timestamp')) ?? new Date().toISOString();

  const { isLoading, isError, data } = useFetchGraphData({
    req: {
      query: {
        eventIds,
        start: `${timestamp}||-30m`,
        end: `${timestamp}||+30m`,
      },
    },
    options: {
      refetchOnWindowFocus: false,
    },
  });
```

This way we can even remove `moment` as well

Reply:
Also as @PhilippeOberti pointed out, some unit tests to check how `useFetchGraphData` is called based on the `getFieldsData` would be important

Reply:
Thanks @opauloh , @PhilippeOberti !
I enhanced the assertions of the current UT and changed the logic to use date math
I believe it is more robust now